src/test/kotlin/eu/europa/ec/eudi/openid4vci/examples/OfferBasedIssuanceUsingAuthorizationFlow.kt

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2023 European Commission
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.europa.ec.eudi.openid4vci.examples
+
+import eu.europa.ec.eudi.openid4vci.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.request.forms.*
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import org.jsoup.Jsoup
+import org.jsoup.nodes.FormElement
+import java.net.URL
+
+private val actingUser = ActingUser("tneal", "password")
+
+fun main(): Unit = runBlocking {
+    val credentialOfferUrl = "eudi-openid4ci://credentialsOffer?credential_offer=%7B%22" +
+        "credential_issuer%22:%22https://dev.issuer-backend.eudiw.dev%22,%22" +
+        "credential_configuration_ids%22:[%22eu.europa.ec.eudiw.pid_mso_mdoc%22," +
+        "%22eu.europa.ec.eudiw.pid_vc_sd_jwt%22,%22org.iso.18013.5.1.mDL%22]," +
+        "%22grants%22:%7B%22authorization_code%22:%7B%22" +
+        "authorization_server%22:%22https://dev.auth.eudiw.dev/realms/pid-issuer-realm%22%7D%7D%7D"
+
+    println("[[Scenario: Issuance based on credential offer url: $credentialOfferUrl]] ")
+
+    val credentialOfferRequestResolver = CredentialOfferRequestResolver(ktorHttpClientFactory = ::httpClientFactory)
+    val credentialOffer = credentialOfferRequestResolver.resolve(credentialOfferUrl).getOrThrow()
+
+    val issuer = Issuer.make(
```

Comment:
Perhaps we can introduce yet another `make` (factory methody of `Issuer`)

Accepting the offered URI and the config. It would resolve the offer and invoke the current `maker`

Reply:
done